packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -146,44 +147,44 @@ export class AuthCodeRedirectHandler implements IRedirectHandler {
         errorIfNull: true,
       }
     )) as string;
-    const isDpop =
-      (await this.storageUtility.getForUser(storedSessionId, "dpop")) ===
-      "true";
 
-    // Since we throw if not found, the type assertion is okay
-    const issuer = (await this.storageUtility.getForUser(
+    const {
+      issuerConfig,
+      codeVerifier,
+      redirectUrl: storedRedirectIri,
+      dpop: isDpop,
+    } = await loadOidcContextFromStorage(
       storedSessionId,
-      "issuer",
-      { errorIfNull: true }
-    )) as string;
+      this.storageUtility,
+      this.issuerConfigFetcher
+    );
 
-    // Store the current session ID specifically in 'localStorage' (i.e., not using
-    // any other storage mechanism), as we don't deem this information to be
-    // sensitive, and we want to ensure it survives a browser tab refresh.
-    window.localStorage.setItem(KEY_CURRENT_SESSION, storedSessionId);
+    if (codeVerifier === undefined) {
+      throw new Error(
+        `The code verifier for session ${storedSessionId} is missing from storage.`
+      );
+    }
+
+    if (storedRedirectIri === undefined) {
```

Comment:
I think we may also need to assert that this is a relative IRI to the current host, such that the redirect IRI can't be tampered with and take you off to `some.evil.host`

Reply:
If it has been tampered with, the subsequent token exchange will fail, because the redirect URL provided as part of the token request must match the redirect URL initially provided during the authorization request, so I don't think we need to add anything specific here.

Reply:
Right, but couldn't the malicious actor pretend to be the AS? i.e., client -> malicious -> AS such that the malicious actor gets back the token?

Reply:
The attack vector in this case would be modifying the `issuerConfig` as you rightly suggested in https://github.com/inrupt/solid-client-authn-js/pull/2009#discussion_r819024793, or modifying the issuer referenced in the user session information. The `storedRedirectIri`is the IRI to which the user should have been redirected already: it's the client-controlled URL where the auth code is returned. It is only verified by the OIDC provider, but at this point the user is not redirected to this URL.